src/main.rs

```diff
@@ -230,3 +200,48 @@ pub fn init_logger() {
             .expect("Could not start logger");
     }
 }
+
+#[cfg(target_os = "windows")]
+fn windows_fix_dpi() {
+    use winapi::shared::windef::DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2;
+    use winapi::um::winuser::SetProcessDpiAwarenessContext;
+    unsafe {
+        SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn handle_macos() {
+    // incase of app bundle, we can just pass --disowned option straight away to bypass this check
+    #[cfg(not(debug_assertions))]
+    if !SETTINGS.get::<CmdLineSettings>().disowned {
+        if let Ok(curr_exe) = std::env::current_exe() {
+            assert!(std::process::Command::new(curr_exe)
+                .args(std::env::args().skip(1))
+                .arg("--disowned")
+                .spawn()
+                .is_ok());
+            return;
```

Comment:
Oh, wait a minute. This was expected to stop execution, but now it doesn't. I'll fix it.